File: Napakalaki/src/napakalaki/BadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author alexnp
 */
public class BadConsequence {
    private String text;
    private int levels;
    private int nVisibleTreasures;
    private int nHiddenTreasures;
    private boolean death;
    private List<TreasureKind> specificVisibleTreasure = new ArrayList<>();
    private List<TreasureKind> specificHiddenTreasure = new ArrayList<>();

    public BadConsequence(String text, int levels, int nVisibleTreasures, int nHiddenTreasures) {
        this.text = text;
        this.levels = levels;
        this.nVisibleTreasures = nVisibleTreasures;
        this.nHiddenTreasures = nHiddenTreasures;
    }

    public BadConsequence(String text, boolean death) {
        this.text = text;
        this.death = death;
    }

    public BadConsequence(String text, int levels, List<TreasureKind> specificVisibleTreasure, List<TreasureKind> specificHiddenTreasure) {
        this.text = text;
        this.levels = levels;
        this.specificVisibleTreasure = (specificVisibleTreasure != null) ? specificVisibleTreasure : new ArrayList<>();
        this.specificHiddenTreasure = (specificHiddenTreasure != null) ? specificHiddenTreasure : new ArrayList<>();
    }

    @Override
    public String toString() {
        return "BadConsequence{" + "text=" + text + ", levels=" + levels + ", nVisibleTreasures=" + nVisibleTreasures + ", nHiddenTreasures=" + nHiddenTreasures + ", death=" + death + ", specificVisibleTreasure=" + specificVisibleTreasure + ", specificHiddenTreasures=" + specificHiddenTreasure + '}';
    }

    public String getText() {
        return text;
    }

    public int getLevels() {
        return levels;
    }

    public int getnVisibleTreasures() {
        return nVisibleTreasures;
    }

    public int getnHiddenTreasures() {
        return nHiddenTreasures;
    }

    public boolean isDeath() {
        return death;
    }

    public List<TreasureKind> getSpecificVisibleTreasure() {
        return specificVisibleTreasure;
    }

    public List<TreasureKind> getSpecificHiddenTreasure() {
        return specificHiddenTreasure;
    }
    
    
}

File: Napakalaki-rb/pkg/Napakalaki-rb-0.0.1/lib/BadConsequence.rb
#encoding: utf-8

class BadConsequence
  attr_accessor :text, :levels, :nVisibleTreasures, :nHiddenTreasures,
    :death, :specificVisibleTreasures, :specificHiddenTreasures

  private_class_method :new
  
  def initialize(text, levels, nVisibleTreasures, nHiddenTreasures, death, specificVisibleTreasures, specificHiddenTreasures)
    @text = text
    @levels = levels
    @nVisibleTreasures = nVisibleTreasures
    @nHiddenTreasures = nHiddenTreasures
    @death = death
    @specificVisibleTreasures = specificVisibleTreasures
    @specificHiddenTreasures = specificHiddenTreasures
  end
  
  def self.new_level_num_treasures(text, levels, nVisibleTreasures, nHiddenTreasures)
    new(text, levels, nVisibleTreasures, nHiddenTreasures, false, Array.new, Array.new)
  end
  
  def self.new_level_specific_treasures(text, levels, specificVisibleTreasures, specificHiddenTreasures)
    new(text, levels, 0, 0, false, specificVisibleTreasures, specificHiddenTreasures)
  end
  
  def self.new_death(text)
    new(text, 0, 0, 0, true, Array.new, Array.new)
  end
  
  def to_s
    "BadConsequence{text=#{@text}, levels=#{@levels}, nVisibleTreasures=#{@nVisibleTreasures}, nHiddenTreasures=#{@nHiddenTreasures}, death=#{@death}, specificVisibleTreasure=#{@specificVisibleTreasures}, specificHiddenTreasures=#{@specificHiddenTreasures}}"
  end
end